winter-framework/src/main/java/de/uni_mannheim/informatik/dws/winter/matching/rules/MatchingRule.java

```diff
@@ -66,80 +81,200 @@ public MatchingRule(double finalThreshold) {
+		this.headerDebugResultsShort.add(this.RECORD2IDENTIFIER);
+
+		this.comparatorLog.addAttribute(this.TOTALSIMILARITY);
+		this.headerDebugResults.add(this.TOTALSIMILARITY);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.COMPARATORNAME);
+		this.headerDebugResultsShort.add(ComparatorLogger.COMPARATORNAME);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.RECORD1VALUE);
+		this.headerDebugResultsShort.add(ComparatorLogger.RECORD1VALUE);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.RECORD2VALUE);
+		this.headerDebugResultsShort.add(ComparatorLogger.RECORD2VALUE);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.RECORD1PREPROCESSEDVALUE);
+		this.headerDebugResultsShort.add(ComparatorLogger.RECORD1PREPROCESSEDVALUE);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.RECORD2PREPROCESSEDVALUE);
+		this.headerDebugResultsShort.add(ComparatorLogger.RECORD2PREPROCESSEDVALUE);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.SIMILARITY);
+		this.headerDebugResultsShort.add(ComparatorLogger.SIMILARITY);
+
+		this.comparatorLogShort.addAttribute(ComparatorLogger.POSTPROCESSEDSIMILARITY);
+		this.headerDebugResultsShort.add(ComparatorLogger.POSTPROCESSEDSIMILARITY);
+
+		this.comparatorToResultLog = new HashMap<Attribute, Attribute>();
+
+	}
+
+	public void addComparatorToLog(Comparator<RecordType, SchemaElementType> comparator) {
+
+		// 4 fix attributes as defined in initialiseMatchingResults().
+		int position = (this.comparatorLog.getSchema().size() - 4) / ComparatorLogger.COMPARATORLOG.length;
+
+		for (Attribute att : ComparatorLogger.COMPARATORLOG) {
+			Attribute schemaAttribute = new Attribute(Integer.toString(position) + '-'
+					+ comparator.getClass().getSimpleName() + '-' + att.getIdentifier());
+			this.comparatorToResultLog.put(schemaAttribute, att);
+			this.comparatorLog.getSchema().add(schemaAttribute);
+			if (!att.getIdentifier().equals(ComparatorLogger.COMPARATORNAME.getIdentifier())) {
+				this.headerDebugResults.add(schemaAttribute);
+			}
+		}
+	}
+
+	public Record initializeDebugRecord(RecordType record1, RecordType record2, int position) {
+
+		String identifier = record1.getIdentifier() + "-" + record2.getIdentifier();
+		if (position != -1) {
+			identifier = Integer.toString(position) + identifier;
+		}
+		Record debug = new Record(identifier);
+		debug.setValue(this.MATCHINGRULE, getClass().getSimpleName());
+		debug.setValue(this.RECORD1IDENTIFIER, record1.getIdentifier());
+		debug.setValue(this.RECORD2IDENTIFIER, record2.getIdentifier());
+
+		return debug;
+	}
+
+	public Record fillDebugRecord(Record debug, Comparator<RecordType, SchemaElementType> comperator, int position) {
+		Iterator<Attribute> schemaIterator = this.comparatorLog.getSchema().get().iterator();
+		ComparatorLogger compLog = comperator.getComparisonLog();
+		while (schemaIterator.hasNext()) {
+			Attribute schemaAtt = schemaIterator.next();
+			if (schemaAtt.getIdentifier()

```

Comment:
please change this to re-use the attributes that were created in addComparatorToLog() instead of re-building the attribute name

Reply:
Done